runtime/compiler/runtime/JITServerAOTDeserializer.cpp

```diff
@@ -640,6 +642,31 @@ JITServerAOTDeserializer::cacheRecord(const WellKnownClassesSerializationRecord
    }
 
 
+bool
+JITServerAOTDeserializer::cacheRecord(const ThunkSerializationRecord *record,
+                                      TR::Compilation *comp, bool &isNew, bool &wasReset)
+   {
+      {
+      OMR::CriticalSection cs(_resetMonitor);
```

Comment:
Nitpick: acquiring the reset monitor is a bit of an overkill, just a read barrier would be enough. Also a comment explaining that we rely on internal synchronization in `{get/set}J2IThunk()` would be helpful.